### PR TITLE
search-blitz: jitter sleep instead of ticker

### PR DESCRIPTION
**Pull request by keegancsmith**
Using a ticker means a query will always run at a specific moment every
minute. By introducing jitter we avoid predictable behaviour. We jitter
by 25% between each query. In practice this means we sleep for 45s ->
1m15s between each request.

The other change here is how long we sleep is affected by how slow the
query we run is. This seems fine, since most queries are very fast and
the jitter is much larger than the query runtime.

A reason against doing this change is we are now less predictable. IE we
can't rule out bad luck (all queries running at the same time). This be
a very good reason to not do this change.

### CODE REVIEW

**Review by tsenart — APPROVED** [review]

File: internal/cmd/search-blitz/main.go [hunk 1]
```diff
@@ -57,20 +57,10 @@ func run(ctx context.Context, wg *sync.WaitGroup) {
...
-		}
-
-		ticker := time.NewTicker(qc.Interval)
-		defer ticker.Stop()
-
-		for {
+    // Randomize start so queries aren't all scheduled at the same time.
+		jitterSleep(ctx, qc.Interval, 1.0)
 
+		for ctx.Err() == nil {
```
Comment (camdencheek) [inline internal/cmd/search-blitz/main.go:63]: I've never seen this pattern before, but that's so much nicer than fitting channel patterns into channel-free code

File: internal/cmd/search-blitz/main.go [hunk 2]
```diff
@@ -57,20 +57,10 @@ func run(ctx context.Context, wg *sync.WaitGroup) {
...
-		case <-ctx.Done():
-			return
-		case <-time.After(randomStart):
-		}
-
-		ticker := time.NewTicker(qc.Interval)
-		defer ticker.Stop()
-
-		for {
+    // Randomize start so queries aren't all scheduled at the same time.
```
Comment (camdencheek) [inline internal/cmd/search-blitz/main.go:60]: Spacing looks off here

**Review by camdencheek — APPROVED** [review]
Looks great. Thanks for catching this

File: internal/cmd/search-blitz/main.go (hunk 2 above)
Comment (keegancsmith) [reply internal/cmd/search-blitz/main.go:60]: will fix in another PR. Got another change I wanna add to search blitz.

**Comment by keegancsmith** [comment]
> With 14 queries, the chance that they ever synchronize is very slim.

Because we run both clients, in effect its 28. But you are right about them synchronising being very unlikely.

> The only thing I can't completely convince myself of is that it makes sense to include the time of the query in the sleep time. This means that longer-running queries will, over time, run fewer times than shorter running queries. This also means we'll get less consistent buckets when rolling the metrics up in prometheus. I don't think it's a problem, but it definitely makes it more difficult to reason about

The rolling up in buckets is a good point. Yeah so this change is completely optional. In general I like using jitter in anything that is polling something else. I also like the readability gains in the code. But, I think I'll abandon since consistency in samples collected is a good point.

**Pull request closed without merging**